static/framework/modules/component.js

```diff
@@ -0,0 +1,320 @@
+			} else {
+				window.Framework.Component.deserializeComponents(node, data);
+			}
+		});
+	}
+
+
+	get data() {
+		let data = {};
+
+		Object.keys(this.tag.properties).forEach((property) => {
+			data[property] = this[property];
+		});
+
+		data.children = window.Framework.Component.serializeComponents(this.view);
+		return data;
+	}
+
+
+	set data(data) {
+		if (typeof data === 'string') {
+			data = JSON.parse(data);
+		}
+
+		Object.keys(data).forEach((property) => {
+			if (property === 'children') {
+				window.Framework.Component.deserializeComponents(this.view, data.children);
+			} else {
+				this[property] = data[property];
+			}
+		});
+	}
+
+
+	__setMappedPropertyValue__(selector, property, value) {
+		[...this.view.querySelectorAll(selector)].forEach((element) => {
+			if (property.startsWith('style.')) {
+				element.style[property.slice('style.'.length)] = value;
+			} else if (property.startsWith('attribute.')) {
+				element.setAttribute(property.slice('attribute.'.length), value);
+			} else {
+				element[property] = value;
+			}
+		});
+	}
+
+
+	__createChangeListener__(name, selector, handlerName) {
+		[...this.view.querySelectorAll(selector)].forEach((element) => {
+
+			const validateProperty = (event) => {
+				this.__validateProperty__(name, event.target.value, handlerName);
+			};
+
+			element.addEventListener('change', validateProperty);
+			/*element.addEventListener('keyup', (event) => {
+				window.setTimeout(() => {
+					validateProperty(event);
+				}, 200);
+			});*/
+
+		});
+	}
+
+
+	__defaultValidateHandler__() {
+	}
+
+
+	__createListeners__() {
+	}
+
+
+	__validateProperty__(name, value, handlerName, maps, mapsSelector, mapsProperty) {
+		this.validators[name].chain.onStateChange = (chain) => {
+			this[this.validators[name].handler || '__defaultValidateHandler__'](chain);
+		};
+		value = window.Framework.Component.toTyped(this.tag.properties[name], value);
+		this.validators[name].chain.validate(value);
+
+		if (typeof this[handlerName || '__defaultHandler__'] === 'function') {
+			const newValue = this[handlerName || '__defaultHandler__'](value, name);
+
+			if (newValue !== undefined) {
+				this[`__${name}__`] = newValue;
+			}
+		} else {
+			this[`__${name}__`] = value;
+		}
+
+		if (maps) {
+			this.__setMappedPropertyValue__(mapsSelector, mapsProperty, value);
+		}
+	}
+
+
+	static createPropertyStub(name, property) {
+		return `
+			get ${name}() {
+				return this.__${name}__;
+			}
+			set ${name}(value) {
+				this.__validateProperty__(
+					'${name}',
+					value,
+					'${property.handlerName}',
+					${property.maps !== undefined},
+					'${(property.maps || {}).selector}',
+					'${(property.maps || {}).property}'
+				);
+			}`;
+	}
+
+	/*
+	 I fully understand the whole depravity of this code, but if anyone could figure out
+	 how to construct an ES6 class in dynamics without eval() please let me know,
+	 or, what's better, change this without breaking everything else.
+	 */
```

Comment:
Hope, it could help you.

```JavaScript
let fieldName = 'marina';

class Marina {
  constructor () {
  	this.marina();
  }

  [fieldName]() {
    console.log('Marina')
  }
}
```

https://learn.javascript.ru/proxy
http://learn.javascript.ru/es-class#геттеры-сеттеры-и-вычисляемые-свойства